ts/Data/DataTable.ts

```diff
@@ -346,15 +346,9 @@ class DataTable implements DataEvent.Emitter {
                 delete columns[columnName];
             }
 
-            let nColumns = Object.keys(columns).length;
-            if (table.rowKeysId && nColumns === 1) {
-                // All columns deleted, remove row keys column
-                delete columns[table.rowKeysId];
-                nColumns = 0;
-            }
-
-            if (!nColumns) {
+            if (!Object.keys(columns).length) {
                 table.rowCount = 0;
+                this.deleteRowIndexReferences();
             }
```

Comment:
Maybe it is time for a `DataTable.clear` function, that keeps columns and only removes rows and index refs.

Reply:
How would it differ from `DataTable.deleteRows`?

I realized that I forgot about this method and didn't implement the references updating if we use `deleteRows` on a modified table.  I'll take care of it right away.